pkg/git/repository.go

```diff
@@ -370,15 +392,14 @@ func (r *repo) addRemote(name, url string) error {
 	return err
 }
 
-func (r *repo) initBranch(ctx context.Context, branchName string) error {
-	w, err := worktree(r)
-	if err != nil {
-		return err
-	}
+func (r *repo) initBranch(ctx context.Context, branchName string) error {	
+	_, w, err := r.commit(&PushOptions{
```

Comment:
also - I am not sure returning the Worktree from the commit func is good. It seems more like a side effect. Maybe it'll be better to duplicate the call to `worktree(r)` (once in `initBranch` and once in `persist`), and send the value into the `commit` func. And commit will only return hash or fail.
I am not %100 sure in this, because this will cause some code duplication. But the way it is currently seems a bit hackey to me.